### PR TITLE
Testing primer

### PR DESCRIPTION
**Pull request by hjhamala**
For review

### CODE REVIEW

File: _posts/2022-10-28-testing-primer-dev-academy.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+* Integration tests are used to find out that components/modules work together
+* E2E tests test the whole application - for a web application this could mean testing the application flow with an automated browser
+For small applications - like Solita assignment - it may be quite hard to make a clear difference between integration and unit especially if the application is not split into clear components.
+
+## Testing approach
+One good approach to testing is through the requirements. The pre-assignment contained the next requirements
+* The journey and the station data should be parsed, validated, and persisted, short trips should be parsed out
+* The journey list should be paginated
+* For each journey show departure and return stations, covered distance in kilometers and duration in minutes
+* top 5 return and departure stations from a station
```
Comment (jarnovayrynen) [inline _posts/2022-10-28-testing-primer-dev-academy.md:36]: Here also a capital letter to start the item

File: _posts/2022-10-28-testing-primer-dev-academy.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+* Tests are quality control. Tests tell us that the things we are developing do the right things - the right way.
+* Tests help find out regressions. Changing the code like refactoring is easier and less nervous when we have confidence that the software works after the changes.
+* Tests are part of the documentation. In many cases, tests can be used to find out how libraries/APIs are expected to be used.
+
+## How we test
+Classically tests are split into three main types of tests.
+* Unit tests test software modules in isolation from each other. 
+* Integration tests are used to find out that components/modules work together
+* E2E tests test the whole application - for a web application this could mean testing the application flow with an automated browser
+For small applications - like Solita assignment - it may be quite hard to make a clear difference between integration and unit especially if the application is not split into clear components.
```
Comment (jarnovayrynen) [inline _posts/2022-10-28-testing-primer-dev-academy.md:29]: Here it seems a  period "." is missing?

**Review by jarnovayrynen — APPROVED** [review]
Added two minor comments how to improve style. Content itself looks nice and the blog contains useful information.

File: _posts/2022-10-28-testing-primer-dev-academy.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+One surprising thing was that there were few or no tests in many applications. Pre-assignment mentioned tests quite clearly. After going through the exercises I found some of the exercises mentioned that an applicant had found difficult to make meaningful tests. 
+ 
+I think that as professional developers we may have forgotten that writing good and useful tests have not been an easy thing to learn - not for me at least. Therefore I present a quick primer based on Solita's fall assignment and how it could be tested. The blog post is mainly aimed at beginner developers. 
+
+## Why do we test?
+First, we should think about the reason why we even bother to write tests. The next three reasons rise to my mind
+* Tests are quality control. Tests tell us that the things we are developing do the right things - the right way.
+* Tests help find out regressions. Changing the code like refactoring is easier and less nervous when we have confidence that the software works after the changes.
+* Tests are part of the documentation. In many cases, tests can be used to find out how libraries/APIs are expected to be used.
```
Comment (janneri) [inline _posts/2022-10-28-testing-primer-dev-academy.md:22]: Tests are users of our own APIs and encourage us to write clear APIs and testable, modular code

File: _posts/2022-10-28-testing-primer-dev-academy.md (hunk 2 above)
Comment (hjhamala) [reply _posts/2022-10-28-testing-primer-dev-academy.md:29]: Seems to miss a line break. "For small application" should be its own paragraph

File: _posts/2022-10-28-testing-primer-dev-academy.md (hunk 3 above)
Comment (hjhamala) [reply _posts/2022-10-28-testing-primer-dev-academy.md:22]: Good point, I moved that to suggestion to different place.

**Pull request merged**